src/error.rs

```diff
@@ -240,3 +275,87 @@ impl fmt::Display for UnalignedVecError {
         write!(f, "vector is unaligned (off by {} bytes)", self.offset)
     }
 }
+
+
+/// Incompatible vector transmutation error.
+///
+/// Returned when the element type `T` does not allow a safe vector
+/// transmutation to the target element type `U`. This happens when either
+/// the size or minimum memory alignment requirements are not met:
+/// 
+/// - `std::mem::align_of::<T>() != std::mem::align_of::<U>()` 
+/// - `std::mem::size_of::<T>() != std::mem::size_of::<U>()`
+#[cfg(feature = "std")]
+#[derive(Clone, Eq, Hash, PartialEq)]
+pub struct IncompatibleVecTargetError<T, U> {
+    /// The original vector.
+    pub vec: Vec<T>,
+    /// The target element type
+    target: PhantomData<U>,
+}
+
+#[cfg(feature = "std")]
+impl<T, U> IncompatibleVecTargetError<T, U> {
+    /// Create an error with the given vector.
+    pub fn new(vec: Vec<T>) -> Self {
+        IncompatibleVecTargetError {
+            vec,
+            target: PhantomData,
+        }
+    }
+
+    /// Create a copy of the data and transmute it. As the new vector will be
+    /// properly aligned for accessing values of type `U`, this operation will
+    /// never fail.
+    ///
+    /// # Safety
+    ///
+    /// The byte data in the vector needs to correspond to a valid contiguous
+    /// sequence of `U` values.
+    pub unsafe fn copy_unchecked(&self) -> Vec<U> {
+        let len = self.vec.len() * core::mem::size_of::<T>() / core::mem::size_of::<U>();
+        let mut out = Vec::with_capacity(len);
+        core::ptr::copy_nonoverlapping(self.vec.as_ptr() as *const u8, out.as_mut_ptr() as *mut u8, len * core::mem::size_of::<U>());
+        out.set_len(len);
+        out
+    }
+
+    /// Create a copy of the data and transmute it. As `T` is trivially
```

Comment:
Why is anything `TriviallyTransmutable`? Are you missing a type bound here?

Reply:
I did miss it! Nice save!